chap_core/geometry.py

```diff
@@ -156,10 +156,32 @@ class Polygons:
     def __init__(self, polygons):
         self._polygons = polygons
 
+    def __eq__(self, other):
+        return self._polygons == other._polygons
+    
+    def __len__(self):
+        return len(self._polygons.features)
+    
+    def __iter__(self):
+        for feat in self._polygons.features:
+            yield feat
+    
+    @property
+    def __geo_interface__(self):
+        return self.to_geojson()
+
     @property
     def data(self) -> FeatureCollectionModel:
         return self._polygons
 
+    @property
+    def bbox(self):
+        from .geoutils import feature_bbox
```

Comment:
Had to do this to avoid circular import with geoutils.py, but not sure if this is in line with how you do this for the rest of chap-core / if you have any other suggestions?